src/client/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -128,18 +133,31 @@ public partial class AuthenticationResult
 
             if (homeAccountId != null)
             {
+                // This method call is to an async method from this constructor but the network call in the method will ideally never happen.
+                var tenantProfiles = Task.Run(async () => await _cacheManager.GetTenantProfilesAsync(homeAccountId).ConfigureAwait(false)).GetAwaiter().GetResult();
```

Comment:
would we want to compute this lazily?
for instance have delegate in account which is called the first time a customer requests tenant profiles?